src/pages/en/guides/migrate-to-astro/from-gatsby.mdx

```diff
@@ -2,67 +2,608 @@
+        frontmatter {
+          date(formatString: "MMMM DD, YYYY")
+          title
+          description
+        }
+      }
+    }
+  }
+`
+```
+
+## Guided example: Gatsby layout to Astro
+
+This example converts the main project layout (`layout.js`) from Gatsby's blog starter to `src/layouts/Layout.astro`.
+
+This page layout shows one header when visiting the home page, and a different header with a link back to Home for all other pages.
+
+1. Identify the return() JSX.
+
+  ```jsx {21-29} title="layout.js"
+  import * as React from "react"
+  import { Link } from "gatsby"
+  const Layout = ({ location, title, children }) => {
+    const rootPath = `${__PATH_PREFIX__}/`
+    const isRootPath = location.pathname === rootPath
+    let header
+    if (isRootPath) {
+      header = (
+        <h1 className="main-heading">
+          <Link to="/">{title}</Link>
+        </h1>
+      )
+    } else {
+      header = (
+        <Link className="header-link-home" to="/">
+          Home
+        </Link>
+      )
+    }
+    return (
+      <div className="global-wrapper" data-is-root-path={isRootPath}>
+        <header className="global-header">{header}</header>
+        <main>{children}</main>
+        <footer>
+          © {new Date().getFullYear()}, Built with
+          {` `}
+          <a href="https://www.gatsbyjs.com">Gatsby</a>
+        </footer>
+      </div>
+    )
+  }
+  export default Layout
+  ```
+
+2. Create `Layout.astro` and add this `return` value, [converted to Astro syntax](#reference-convert-to-astro-syntax). 
+
+  Note that:
+
+  - `{new Date().getFullYear()}` just works 🎉
+  - `{children}` becomes `<slot />` 🦥
+  - `className` becomes `class` 📛
+  - `Gatsby` becomes `Astro` 🚀
+
+  ```astro title="src/layouts/Layout.astro" "<slot />" "class" "Astro" "astro.build"
+  ---
+  ---
+  <div class="global-wrapper" data-is-root-path={isRootPath}>
+    <header class="global-header">{header}</header>
+    <main><slot /></main>
+    <footer>
+      © {new Date().getFullYear()}, Built with
+      {` `}
+      <a href="https://www.astro.build">Astro</a>
+    </footer>
+  </div>
+  ```
+
+3. Add a page shell so that your layout provides each page with the necessary parts of an HTML document:
+
+  ```astro title="src/layouts/Layout.astro" ins={3-10,22-23}
+  ---
+  ---
+  <html>
+    <head>
+      <meta charset="utf-8" />
+      <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+      <meta name="viewport" content="width=device-width" />
+      <title>Astro</title>
+    </head>
+    <body>
+      <div class="global-wrapper" data-is-root-path={isRootPath}>
+        <header class="global-header">{header}</header>
+        <main>
+          <slot />
+        </main>
+        <footer>
+          &#169; {new Date().getFullYear()}, Built with
+          {` `}
+          <a href="https://www.astro.build">Astro</a>
+        </footer>
+      </div>
+    </body>
+  </html>
+  ```
+
+3. Add any needed JavaScript, props, imports.
```

Comment:
This is mentioned several times but in different order and styling (comma, no comma, with/without "and"), so I added a few suggestions to make them all the same.

```suggestion
3. Add any needed imports, props, and JavaScript
```